js_modules/dagit/packages/core/src/workers/util.ts

```diff
@@ -0,0 +1,25 @@
+// https://developer.chrome.com/blog/how-to-convert-arraybuffer-to-and-from-string/
+
+export function arrayBufferToString(buf: ArrayBuffer): string {
+  let binary = '';
+  const bytes = new Uint16Array(buf);
+  const len = bytes.byteLength;
+  for (let i = 0; i < len; i++) {
+    const char = String.fromCharCode(bytes[i]);
+    if (char !== '\x00') {
+      binary += char;
+    }
+  }
```

Comment:
Hmm this seems like it could be a pretty big bottleneck, is there a particular field in our JSON that contains null characters? It seems like removing them during both serialization and deserialization is unnecessary. (Nice for completeness, but if they're always used together it'd be safe to clean only the serializer?)

If we could reduce this entire function to this line, I think it'd be a big win because we can avoid having to concatenate onto a string over and over:

```
String.fromCharCode.apply(null, new Uint16Array(buf))
```